docs/how-tos/how-to-use-searchkick-for-search-autocomplete.mdx

```diff
@@ -0,0 +1,285 @@
+---
+sidebar_position: 10
+---
+
+# How to use Searchkick for search autocomplete
+
+In this guide, we'll see how to improve the autocomplete capabilities
+offered by default in [Solidus Starter Frontend][ssf], providing support for large
+dataset and advanced features like full-text search, typo tolerance, etc.
+
+This how-to will use a gem called [Searchkick][searchkick-github], which
+is one of most popular choices in the Rails ecosystem.
+
+## Installing Searchkick
+
+Let's get started by adding the necessary gems to our Gemfile:
+
+```
+bundle add searchkick elasticsearch
+```
+
+:::info
+
+Searchkick supports both Elasticsearch and OpenSearch. This guide
+section covers the installation with Elasticsearch. Please, refer to
+[Searchkick README][searchkick-github] for more installation options.
+
+:::
+
+## Indexing Configuration
+
+We can now customize our Product model to tell it how its information
+should be indexed. We can [customize][override-core] the `Spree::Product`
+core model:
+
+```ruby title="app/overrides/my_store/spree/product/add_searchkick.rb"
+module AmazingStore
+  module Spree
+    module Product
+      module AddSearchkick
+        def self.prepended(base)
+          unless base.respond_to?(:searchkick_index)
```

Comment:
Why do we need that check?

Reply:
Because Searchkick does not allow to call `searchkick` multiple times on the same model; there are several issues reported with similar issues (eg. https://github.com/ankane/searchkick/issues/596, https://github.com/ankane/searchkick/issues/313 and https://github.com/ronzalo/spree_searchkick/issues/7). This happens when the application gets reloaded and this override is re-applied to the class it is prepended to. One alternative might be having a special override that is not reloaded, but I think it makes it even more complex. Do you have other solutions in mind?

Reply:
Thanks for your clarification. I guess there's no better workaround. Would a small comment about the issue be worth it?